packages/truffle-compile/compilerSupplier/loadingStrategies/Native.js

```diff
@@ -0,0 +1,57 @@
+const { execSync } = require("child_process");
+const ora = require("ora");
+const LoadingStrategy = require("./LoadingStrategy");
+
+class Native extends LoadingStrategy {
+  async getSolc(commitString) {
+    const solcFileName = this.getCachedSolcFileName(commitString);
+    if (solcFileName) return this.getCachedSolcByFileName(solcFileName);
+
+    const allVersions = await this.getSolcVersions(this.config.versionsUrl);
+    const fileName = this.getVersionUrlSegment(commitString, allVersions);
+
+    if (!fileName) throw new Error("No matching version found");
+
+    const url = this.config.compilerUrlRoot + fileName;
+    const spinner = ora({
+      text: "Downloading compiler",
+      color: "red"
+    }).start();
+
+    return this.getSolcByUrlAndCache(url, fileName, spinner);
+  }
```

Comment:
(Same comment as with Docker: can we move this logic into VersionRange so that Native just uses that?)